wavenet.py

```diff
@@ -152,9 +166,10 @@ def predict_proba(self, waveform, name='wavenet'):
 
     def loss(self, input_batch, name='wavenet'):
         with tf.variable_scope(name):
-            input_batch = self._preprocess(input_batch)
-            encoded = self._one_hot(input_batch)
-            raw_output = self._create_network(encoded)
+            quantized = self._preprocess(input_batch)
+            encoded = self._one_hot(quantized)
+            inputs = tf.reshape(input_batch, [self.batch_size, -1, 1])
+            raw_output = self._create_network(inputs)
```

Comment:
Have we tested this to see if it helps with generation? I'd suggest moving this to a separate PR, as it's quite a big change in the network architecture.